api_controller/models.py

```diff
@@ -0,0 +1,139 @@
+from django.db import models
+from django.conf import settings
+from translation.models import TranslatedLesson, TranslationStep
+from translation.models import YandexTranslator
+
+import requests
+
+
+class ApiController(models.Model):
+    api_key = 0
+    oauth_credentials = {
+        "client_id": settings.STEPIK_CLIENT_ID,
+        "client_secret": settings.STEPIK_CLIENT_SECRET,
+    }
+    base_url = models.TextField()
+    api_version = 0.1
+    api_host = "https://stepik.org"
+    token = None
+
+    def stepik_oauth(self):
+
+        auth = requests.auth.HTTPBasicAuth(self.oauth_credentials["client_id"], self.oauth_credentials["client_secret"])
+        response = requests.post('https://stepik.org/oauth2/token/',
+                                 data={'grant_type': 'client_credentials'},
+                                 auth=auth)
+        self.token = response.json().get('access_token', None)
+        print(self.token)
+        if not self.token:
+            print('Unable to authorize with provided credentials')
+            exit(1)
+
+    def fetch_stepik_object(self, obj_class, obj_id):
+        api_url = '{}/api/{}s/{}'.format(self.api_host, obj_class, obj_id)
+        print("API_URL", api_url)
+        response = requests.get(api_url,
+                                headers={'Authorization': 'Bearer ' + self.token}).json()
+        if '{}s'.format(obj_class) in response:
+            return response['{}s'.format(obj_class)][0]
+        return None
+
+    def fetch_stepik_step(self, pk):
+        return self.fetch_stepik_object("step", pk)
+
+    def fetch_stepik_lesson(self, pk):
+        return self.fetch_stepik_object("lesson", pk)
+
+    def steps_text(self, ids):
+        texts = []
+        for id in ids:
+            step = self.fetch_stepik_object("step", id)
```

Comment:
We can fetch all the steps with multiple parameter "ids[]" in the query api/steps?ids[]=1&ids[]=2. Do not use single requests here, because it can affect performance significantly. Also, we can use api/steps?lesson=%lessonId%, it also will return all variations of each step.